src/ansys/aedt/core/hfss.py

```diff
@@ -3803,59 +3811,93 @@ def edit_sources(
-        argument = []
-        if self.solution_type == "Terminal":
-            argument.extend(["UseIncidentVoltage:=", use_incident_voltage])
+                    power = vals[0]
+                    if len(vals) == 1:
+                        phase = "0deg"
+                    else:
+                        phase = vals[1]
+                if isinstance(vals, (list, tuple)) and len(vals) == 3:
+                    terminated = vals[2]
+                else:
+                    terminated = False
+                if use_incident_voltage and self.solution_type == "Terminal":
+                    setting.append(["Name:=", key, "Terminated:=", terminated, "Magnitude:=", power, "Phase:=", phase])
+                else:
+                    setting.append(["Name:=", key, "Magnitude:=", power, "Phase:=", phase])
+            argument = []
+            if self.solution_type == "Terminal":
+                argument.extend(["UseIncidentVoltage:=", use_incident_voltage])
 
-        argument.extend(
-            [
-                "IncludePortPostProcessing:=",
-                include_port_post_processing,
-                "SpecifySystemPower:=",
-                True if max_available_power else False,
-            ]
-        )
+            argument.extend(
+                [
+                    "IncludePortPostProcessing:=",
+                    include_port_post_processing,
+                    "SpecifySystemPower:=",
+                    True if max_available_power else False,
+                ]
+            )
 
-        if max_available_power:
-            argument.append("Incident Power:=")
-            argument.append(max_available_power)
+            if max_available_power:
+                argument.append("Incident Power:=")
+                argument.append(max_available_power)
+            args = [argument]
+            args.extend(setting)
+        else:
+            eigenmode_type_definition = "EigenStoredEnergy" if eigenmode_stored_energy else "EigenPeakElectricField"
+            argument = ["FieldType:=", eigenmode_type_definition]
+            power = []
+            phase = []
+            for _, value in assignment.items():
+                if eigenmode_stored_energy:
+                    power.append(value)
+                else:
+                    if isinstance(value, (list, tuple)):
+                        power.append(value[0])
+                        phase.append(value[1])
+                    elif isinstance(value, str):
+                        power.append(value)
+                        phase.append("0deg")
+            if eigenmode_stored_energy:
+                setting = ["Name:=", "Modes", "Magnitudes:=", power]
+            else:
+                setting = ["Name:=", "Modes", "Magnitudes:=", power, "Phases:=", phase]
 
-        args = [argument]
-        args.extend(setting)
+            args = [argument]
+            args.extend([setting])
```

Comment:
Could settings be defined as a list of list of every cases, i.e. for "Eigenmode" and non "Eigenmode"?
This way we could move `args = [argument]` and `args.extend(settings)` at the same level as `self.osolution.EditSources(args)`. I think this would avoid code duplicate and help for future maintenance / readability.